msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CodesignTaskBase.cs

```diff
@@ -110,11 +164,11 @@ public override bool Execute ()
 			if (Resources.Length == 0)
 				return true;
 
-			EnvironmentVariables = new string[] {
-				"CODESIGN_ALLOCATE=" + CodesignAllocate
-			};
+			Parallel.ForEach (Resources, new ParallelOptions { MaxDegreeOfParallelism = 4 }, (item) => {
```

Comment:
`= 4` might not be the adequate in every case. Maybe it should be something based on`Environment.ProcessorCount` ?

A bit tricky to get a number since *right now* the code is likely I/O bound, i.e. waiting for the network request for a timestamp.

However if/when we add `--timestamp=none`, e.g. for debug builds it should *always* be safe, then `codesign` will become CPU bound.